### PR TITLE
fix: bugfix for smart pay

### PR DESCRIPTION
**Pull request by nuanyang233**
## Description

Closes #MF-3075 MF-3135 MF-3136 MF-3137 MF-3140 MF-3144 MF-3145 MF-3146 MF-3147 MF-3148 MF-3149 MF-3150 MF-3151 MF-3154 MF-3175

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/web3-shared/evm/src/types/index.ts [hunk 1]
```diff
@@ -266,6 +266,7 @@ export interface Transaction {
 
     // 4337
     gasCurrency?: string
+    is_fund?: boolean
```
Comment (guanbinrui) [inline packages/web3-shared/evm/src/types/index.ts]:
```suggestion
    isFund?: boolean
```

File: packages/mask/src/extension/popups/pages/Wallet/ContractInteraction/index.tsx [hunk 2]
```diff
@@ -170,6 +170,7 @@ const ContractInteraction = memo(() => {
         })
     }, [TransactionFormatter, request])
 
+    console.log(request)
```
Comment (guanbinrui) [inline packages/mask/src/extension/popups/pages/Wallet/ContractInteraction/index.tsx]:
```suggestion
```

File: packages/mask/src/extension/popups/pages/Wallet/Transfer/Transfer1559.tsx [hunk 3]
```diff
@@ -382,7 +382,7 @@ export const Transfer1559 = memo<Transfer1559Props>(({ selectedAsset, openAssetM
         EthereumAddress.isValid(address) ? address : registeredAddress,
     )
     // #endregion
-
+    console.log(minGasLimit)
```
Comment (guanbinrui) [inline packages/mask/src/extension/popups/pages/Wallet/Transfer/Transfer1559.tsx]:
```suggestion
```

File: packages/plugins/EVM/src/state/Connection/interceptors/ContractWallet.ts [hunk 4]
```diff
@@ -91,11 +91,6 @@ export class ContractWallet implements Middleware<Context> {
         throw new Error('No user operation to be estimated.')
     }
 
-    private async fund(context: Context) {
```
Comment (guanbinrui) [inline packages/plugins/EVM/src/state/Connection/interceptors/ContractWallet.ts]: Revert this.

File: packages/plugins/EVM/src/state/Connection/interceptors/MaskWallet.ts [hunk 5]
```diff
@@ -22,6 +30,13 @@ export class MaskWallet implements Middleware<Context> {
                     break
                 }
                 break
+            case EthereumMethodType.MASK_FUND:
```
Comment (guanbinrui) [inline packages/plugins/EVM/src/state/Connection/interceptors/MaskWallet.ts]: Revert this.

File: packages/plugins/EVM/src/state/Connection/middleware/RecentTransaction.ts [hunk 6]
```diff
@@ -17,14 +18,10 @@ export class RecentTransaction implements Middleware<Context> {
...
                 case EthereumMethodType.MASK_FUND:
-                    if (!context.config || typeof context.result !== 'string') return
-                    await Transaction?.addTransaction?.(
-                        context.chainId,
-                        context.account,
-                        context.result,
-                        context.config,
-                    )
-                    await TransactionWatcher?.watchTransaction(context.chainId, context.result, context.config)
+                    const tx = (context.result as FunderAPI.Fund).message.tx
```
Comment (guanbinrui) [inline packages/plugins/EVM/src/state/Connection/middleware/RecentTransaction.ts]: Do it in `packages/plugins/EVM/src/state/Connection/interceptors/ContractWallet.ts`. The `MASK_FUND` is expected to return a tx hash.

File: packages/plugins/SmartPay/src/hooks/useDeploy.tsx [hunk 7]
```diff
@@ -77,49 +86,36 @@ export function useDeploy(
...
+        closeSnackbar()
 
         try {
-            const response = await SmartPayFunder.fund(chainId, {
-                publicKey,
-                type: signPersona ? ProofType.Persona : ProofType.EOA,
-                signature,
-                payload,
-            })
+            const response = await connection?.fund?.(
```
Comment (guanbinrui) [inline packages/plugins/SmartPay/src/hooks/useDeploy.tsx]: It should return a tx hash.

File: packages/plugins/EVM/src/state/Connection/middleware/RecentTransaction.ts (hunk 6 above)
Comment (nuanyang233) [reply packages/plugins/EVM/src/state/Connection/middleware/RecentTransaction.ts]: No, `MASK_FUND` needs to return the wallet address in addition to the tx hash.

File: packages/web3-providers/src/SmartPay/apis/SmartPayFunderAPI.ts [hunk 8]
```diff
@@ -4,7 +4,7 @@ import { EMPTY_LIST, Proof } from '@masknet/shared-base'
 import { FunderAPI } from '../../types/Funder.js'
 import { Web3API } from '../../EVM/index.js'
 import { fetchJSON } from '../../entry-helpers.js'
-import { FUNDER_PROD } from '../constants.js'
+import { FUNDER_DEV } from '../constants.js'
```
Comment (guanbinrui) [inline packages/web3-providers/src/SmartPay/apis/SmartPayFunderAPI.ts]: DEV?

File: packages/web3-providers/src/SmartPay/apis/SmartPayBundlerAPI.ts [hunk 9]
```diff
@@ -4,18 +4,18 @@ import type { ChainId, UserOperation } from '@masknet/web3-shared-evm'
 import { toBase64, fromHex } from '@masknet/shared-base'
 import type { BundlerAPI } from '../../types/Bundler.js'
 import { fetchJSON } from '../../entry-helpers.js'
-import { BUNDLER_PROD } from '../constants.js'
+import { BUNDLER_DEV } from '../constants.js'
```
Comment (guanbinrui) [inline packages/web3-providers/src/SmartPay/apis/SmartPayBundlerAPI.ts]: DEV?

File: packages/web3-providers/src/helpers/fetchCached.ts [hunk 10]
```diff
@@ -25,6 +25,9 @@ const RULES = {
     'https://chainbase-proxy.r2d2.to/v1/ens': Duration.SHORT,
     // Mask Search List
     'https://dsearch.mask.r2d2.to': Duration.LONG,
+    // Smart Pay
+    'https://9rh2q3tdqj.execute-api.ap-east-1.amazonaws.com/': Duration.SHORT,
```
Comment (guanbinrui) [inline packages/web3-providers/src/helpers/fetchCached.ts]:
```suggestion
    'https://9rh2q3tdqj.execute-api.ap-east-1.amazonaws.com': Duration.SHORT,
```

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**